hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/ApproveAllowanceValidator.java

```diff
@@ -181,6 +181,10 @@ private void validateNftAllowances(
             final var spenderAccount = accountStore.getAccountById(spender);
             validateNFTSpender(serialNums, spenderAccount);
 
+            if (Boolean.TRUE.equals(allowance.approvedForAll())) {
```

Comment:
is the logic actually (I _think_ this is equivalent, but clearer):

```
if (allowance.approvedForAll()) {
   validateNFTSpender(spenderAccount); 
} else {
   validateNFTSpender(serialNums, spenderAccount); 
}
```

Reply:
Thanks for the feedback. Unfortunately, we cannot simplify it like that because the return type is the "Boolean" wrapper class. Thus, it could produce a NullPointerException if it's unboxed and is null. I also agree that we can optimize it with an else statement. :) I've modified it to:

     if (Boolean.TRUE.equals(allowance.approvedForAll())) {
                validateNFTSpender(spenderAccount);
            } else {
                validateNFTSpender(serialNums, spenderAccount);
            }
            
            This way, we don't call validateNFTSpender(serialNums, spenderAccount); when we don't have to.